src/expr/node_manager.h

```diff
@@ -578,12 +578,22 @@ class NodeManager {
   /**
   * Create a singleton set from the given element n.
   * @param t the element type of the returned set.
-  * Note that the type of n needs to be a subtype of t.
+  *          Note that the type of n needs to be a subtype of t.
   * @param n the single element in the singleton.
   * @return a singleton set constructed from the element n.
   */
   Node mkSingleton(const TypeNode& t, const TNode n);
 
+  /**
+  * Create a bag from the given element n along with its multiplicity m.
+  * @param t the element type of the returned bag.
+  *          Note that the type of n needs to be a subtype of t.
+  * @param n the element that is used to to construct the bag
+  * @param m the multiplicity of the element n
+  * @return a bag that contains m occurrences of n.
+  */
+  Node mkBag(const TypeNode& t, const TNode n, const TNode m);
```

Comment:
The right name for this method is probably (and unfortunately) `mkMkBag`, since you are asking to make a node of MK_BAG kind.

Reply:
We need a better name. Or another reason to use mkOp<T>(Sort s) :)

Reply:
`buildMKBag`, `buildMakeBag`, `constructMkBag`, `createMkBag`. 
Actually MkBag itself is ugly.

Reply:
mkMkBag is consistent with our naming scheme, either that or we should rename MK_BAG.